src/System.Private.CoreLib/shared/System/Resources/ResourceReader.cs

```diff
@@ -743,7 +758,81 @@ private object _LoadObjectV2(int pos, out ResourceTypeCode typeCode)
             }
 
             // Normal serialized objects
-            throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
+            int typeIndex = typeCode - ResourceTypeCode.StartOfUserTypes;
+            return DeserializeObject(typeIndex);
+        }
+
+        // Helper method to deserialize a type
+        private Object DeserializeObject(int typeIndex)
+        {
+            if (!_permitDeserialization)
+            {
+                throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
+            }
+
+            if (_binaryFormatter == null)
+            {
+                InitializeBinaryFormatter();
+            }
+
+            Type type = FindType(typeIndex);
+ 
+            // Ensure that the object we deserialized is exactly the same
+            // type of object we thought we should be deserializing. 
+            Object graph;
+            graph = s_deserializeMethod(_binaryFormatter, _store.BaseStream);
+            
+            // This check is about correctness, not security at this point.
```

Comment:
I'm not sure this comment adds value.  Consider deleting it or replacing it with something like "Ensure that the deserialized type matches the expected type, in order to minimize the chances of returning corrupted data."

Reply:
Sure, copied this from desktop